pynest/CMakeLists.txt

```
@@ -82,6 +82,7 @@ if ( HAVE_PYTHON )
   install( CODE "
       execute_process(
           COMMAND ${PYTHON} ${CMAKE_CURRENT_BINARY_DIR}/setup.py install_egg_info
+            --root=\${DESTDIR}/
```

My question is if the "/" is considered an empty argument. When `DESTDIR` is empty it reads `--root=/`
```suggestion
            --root=\"\${DESTDIR}\"
```
(I did some quote-hunting a while ago, and possibly missing quotes around `CMAKE_CURRENT_BINARY_DIR` jump to my eye. Might be necessary if there is a space somewhere in the install path… :grimacing: )

I don't have much time to actually try this now, but @jougs and @Helveg where building the wheel recently and might be able to comment on best practice here.

---

The suggestion looks good. I'd also add quotes around `${CMAKE_CURRENT_BINARY_DIR}/setup.py' like @terhorstd mentions.

---

`--root=/` will install to the root directory. When the option is not specified, it installs to the root directory.

---

yes, but it should install into the venv or conda env or wherever the current python thinks it's packages should go.

---

`--root` specifies an additional staging prefix, the actual installation path is relative to this prefix.

---

> The suggestion looks good. I'd also add quotes around `${CMAKE_CURRENT_BINARY_DIR}/setup.py` like @terhorstd mentions.

Is this just based on a gut feeling, or have you actually verified this is necessary?

execute_process already adds quoting to commands, so this would just add noise. Also, the `${PYTHON}` interpreter path may contain spaces as well.

---

I can confirm that `python my path/package/setup.py install_egg_info` fails, but I haven't tried it in this context:

```
(bsb3py9) robin@LAPTOP-JF6T3PSU:~/git$ python bsb hdf5/setup.py install_egg_info
/home/robin/.pyenv/versions/bsb3py9/bin/python: can't find '__main__' module in '/mnt/c/Users/pwd06/git/bsb'
```

---

> execute_process already adds quoting

When **you**  don't use execute_process, **you** have to add the quoting.

---

Do you mean that:

```
execute_process(
          COMMAND ${PYTHON} ${CMAKE_CURRENT_BINARY_DIR}/setup.py install_egg_info
```

with `${CMAKE_CURRENT_BINARY_DIR}` equal to `my path` would be escaped to:

```
python "my path/setup.py" install_egg_info
```

That seems advanced.

---

So is this resolved? @StefanBruens I still think we need the quotes for when there's whitespaces in the paths